backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java

```diff
@@ -380,23 +380,20 @@ public Database setSettings(Map<String, String> settings) {
   public User addUser(String userName) {
     if (!hasUser(userName)) {
       long start = System.currentTimeMillis();
-      // need elevated privileges, so clear user and run as root
-      // this is not thread safe therefore must be in a transaction
-      tx(
-          db -> {
-            String currentUser = db.getActiveUser();
-            try {
-              db.becomeAdmin();
-              executeCreateUser((SqlDatabase) db, userName);
-            } finally {
-              db.setActiveUser(currentUser);
-            }
-          });
+      // only root users can do this
+      executeCreateUser(getAdminJooq(), userName);
       log(start, "created user " + userName);
     }
     return getUser(userName);
   }
 
+  public DSLContext getAdminJooq() {
```

Comment:
public ?

Reply:
ah, yes, that is too much. Should be package private.